From 1c1a4ded2e169b8e0e3ba128d61143e6d7f5f479 Mon Sep 17 00:00:00 2001
From: Howard Butler <hobu.inc@gmail.com>
Date: Wed, 31 Jul 2024 12:45:23 -0500
Subject: [PATCH] protoc detection

---
 cmake/FindProtobufC.cmake | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/cmake/FindProtobufC.cmake b/cmake/FindProtobufC.cmake
index d718ad1c4..a037452ed 100644
--- a/cmake/FindProtobufC.cmake
+++ b/cmake/FindProtobufC.cmake
@@ -90,13 +90,13 @@ find_library(PROTOBUFC_LIBRARY
 mark_as_advanced(PROTOBUFC_LIBRARY)
 
 find_path(PROTOBUFC_INCLUDE_DIR
-							NAMES "google/protobuf-c/protobuf-c.h"
+							NAMES "protobuf-c.h" "google/protobuf-c/protobuf-c.h"
 							PATHS "/usr" "/usr/local" "/opt" ENV PROTOBUFC_ROOTDIR
 							PATH_SUFFIXES "include")
 mark_as_advanced(PROTOBUFC_INCLUDE_DIR)
 
 find_program(PROTOBUFC_COMPILER
-							NAMES "protoc-c"
+							NAMES "protoc" "protoc-c"
 							PATHS "/usr" "/usr/local" "/opt" ENV PROTOBUFC_ROOTDIR
 							PATH_SUFFIXES "bin")
 mark_as_advanced(PROTOBUFC_COMPILER)
