From 0dea36d9e843c2a897d8c0feba5daf2e7203cc30 Mon Sep 17 00:00:00 2001
From: Howard Butler <hobu.inc@gmail.com>
Date: Tue, 9 Jul 2024 10:21:34 -0500
Subject: [PATCH] CMakeLists.txt: define _XKEYCHECK_H to avoid build issue with
 older MSVC SDKs

---
 CMakeLists.txt | 20 ++++++++++++--------
 1 file changed, 12 insertions(+), 8 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d42e56bf0..d2fbb4355 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -273,10 +273,10 @@ src/renderers/agg/src/clipper.cpp
 include_directories(src/renderers/agg/include)
 
 set(v8_SOURCES
-  src/mapscript/v8/v8_object_wrap.hpp  
+  src/mapscript/v8/v8_object_wrap.hpp
   src/mapscript/v8/point.cpp
   src/mapscript/v8/line.cpp
-  src/mapscript/v8/shape.cpp  
+  src/mapscript/v8/shape.cpp
   src/mapscript/v8/v8_mapscript.cpp
   src/mapv8.cpp
 )
@@ -311,7 +311,7 @@ else(WIN32)
   set(REGEX_SOURCES "")
 endif(WIN32)
 
-set(mapserver_SOURCES src/fontcache.c 
+set(mapserver_SOURCES src/fontcache.c
 src/cgiutil.c src/mapgeos.c src/maporaclespatial.c src/mapsearch.c src/mapwms.cpp src/classobject.c
 src/mapgml.c src/mapoutput.c src/mapwmslayer.c src/layerobject.c src/mapgraticule.c src/mapows.c src/mapogcapi.cpp
 src/mapservutil.c src/mapxbase.c src/maphash.c src/mapowscommon.c src/mapshape.c src/mapxml.c src/mapbits.c
@@ -378,7 +378,7 @@ if(BUILD_DYNAMIC)
   set_target_properties( mapserver  PROPERTIES
     VERSION ${MapServer_VERSION_STRING}
     SOVERSION 2
-) 
+)
 endif(BUILD_DYNAMIC)
 
 if(BUILD_STATIC)
@@ -386,7 +386,7 @@ if(BUILD_STATIC)
   set_target_properties( mapserver_static PROPERTIES
     VERSION ${MapServer_VERSION_STRING}
     SOVERSION 2
-  ) 
+  )
 endif(BUILD_STATIC)
 
 #SOVERSION is not necessarily the same as the
@@ -421,9 +421,9 @@ add_executable(shptreetst src/apps/shptreetst.c)
 target_link_libraries(shptreetst ${MAPSERVER_LIBMAPSERVER})
 endif()
 
-if (CMAKE_BUILD_TYPE STREQUAL "Debug") 
+if (CMAKE_BUILD_TYPE STREQUAL "Debug")
   set(USE_EXTENDED_DEBUG 1)
-endif (CMAKE_BUILD_TYPE STREQUAL "Debug") 
+endif (CMAKE_BUILD_TYPE STREQUAL "Debug")
 
 find_package(PNG)
 if(PNG_FOUND)
@@ -797,7 +797,7 @@ if(WITH_XMLMAPFILE)
    if(NOT USE_LIBXML2)
     report_dependency_error(KML LIBXML2)
    endif(NOT USE_LIBXML2)
-  
+
   #check for xslt and exslt
   find_package(LibXslt)
   if(LIBXSLT_FOUND)
@@ -1017,6 +1017,10 @@ include_directories("${PROJECT_BINARY_DIR}")
 if(WIN32)
 set(REGEX_MALLOC 1)
 set(USE_GENERIC_MS_NINT 1)
+
+# FIXME: https://stackoverflow.com/a/65321305 this handles older MSVC
+# SDKs that are used by Conda Forge windows builds
+add_definitions(-D_XKEYCHECK_H=1)
 endif(WIN32)
 
 
