From bb63aacef2e0aa43e4de1a8dbd48fc2aa8f8587f Mon Sep 17 00:00:00 2001
From: Howard Butler <hobu.inc@gmail.com>
Date: Fri, 2 Aug 2024 17:00:50 -0500
Subject: [PATCH 2/2] support setting PROTOBUF_PROTOC_EXECUTABLE for cross
 compilation scenarios

---
 cmake/FindProtobufC.cmake | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/cmake/FindProtobufC.cmake b/cmake/FindProtobufC.cmake
index a037452ed..789d772b4 100644
--- a/cmake/FindProtobufC.cmake
+++ b/cmake/FindProtobufC.cmake
@@ -55,6 +55,7 @@
 
 set(PROTBUFC_SOURCE_EXTENSION "pb-c.c")
 set(PROTBUFC_HEADER_EXTENSION "pb-c.h")
+
 function(PROTOBUFC_GENERATE_C SOURCES HEADERS)
 	if(NOT ARGN)
 		message(SEND_ERROR "Error: PROTOBUFC_GENERATE_C() called without any proto files")
@@ -101,6 +102,11 @@ find_program(PROTOBUFC_COMPILER
 							PATH_SUFFIXES "bin")
 mark_as_advanced(PROTOBUFC_COMPILER)
 
+
+IF (DEFINED PROTOBUF_PROTOC_EXECUTABLE)
+  SET(PROTOBUFC_COMPILER "${PROTOBUF_PROTOC_EXECUTABLE}")
+ENDIF (DEFINED PROTOBUF_PROTOC_EXECUTABLE)
+
 include(FindPackageHandleStandardArgs)
 FIND_PACKAGE_HANDLE_STANDARD_ARGS(ProtobufC DEFAULT_MSG PROTOBUFC_LIBRARY PROTOBUFC_COMPILER PROTOBUFC_INCLUDE_DIR )
 
