From 0100095b1d2186c4fb11f2f8f95135bd31530afb Mon Sep 17 00:00:00 2001
From: Howard Butler <hobu.inc@gmail.com>
Date: Tue, 9 Jul 2024 10:33:18 -0500
Subject: [PATCH 57/60] CMakeLists.txt: need zstd if we're WIN32 and asking for
 mysql

---
 CMakeLists.txt | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5ca7f553e..da3e95a67 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -569,6 +569,11 @@ if(WITH_MYSQL)
     ms_link_libraries( ${MYSQL_LIBRARY})
     list(APPEND ALL_INCLUDE_DIRS ${MYSQL_INCLUDE_DIR})
     set (USE_MYSQL 1)
+    if (WIN32)
+        find_package(Zstd REQUIRED CONFIG)
+        include_directories(${ZSTD_INCLUDE_DIRS})
+        ms_link_libraries(zstd::libzstd_shared)
+    endif (WIN32)
   else(MYSQL_FOUND)
     report_optional_not_found(MYSQL)
   endif(MYSQL_FOUND)
